later/lib/database/datasources/records_local_data_source.dart

```diff
@@ -36,13 +35,49 @@ class RecordsLocalDataSource extends RecordsDataSource {
     return record;
   }
 
+  @override
+  Future<RecordDbModel> update(RecordDbModel recordDbModel) async {
+    final dbModel = await _isar.recordDbModels.get(recordDbModel.id);
+
+    if (dbModel != null) {
+      dbModel
+        ..title = recordDbModel.title == dbModel.title
+            ? dbModel.title
+            : recordDbModel.title
+        ..url =
+            recordDbModel.url == dbModel.url ? dbModel.url : recordDbModel.url
+        ..description = recordDbModel.description == dbModel.description
+            ? dbModel.description
+            : recordDbModel.description
+        ..lastEditedAt = DateTime.now();
+    } else {
+      throw const FormatException(
+          'We can not update file that we don\'t have at Database.');
+    }
+
+    await _isar.writeTxn(() => _isar.recordDbModels.put(dbModel));
+
+    return dbModel;
+  }
+
   @override
   Future<RecordDbModel?> getById(int id) {
-    return _isar.txn(() => _isar.recordDbModels.get(id));
+    final model = _isar.txn(() => _isar.recordDbModels.get(id));
+
+    if (model == null) {
+      throw const FormatException('Object with that id don\'t exist.');
+    }
+    return model;
   }
 
   @override
   Stream<List<RecordDbModel>> watchAll() {
     return _isar.recordDbModels.where().watch(fireImmediately: true);
   }
+
+  @override
+  Future<String> delete(int id) async {
+    final result = await _isar.recordDbModels.delete(id);
+    return 'Resul success:$result';
+  }
```

Comment:
Why do we return a string here?

Reply:
It seemed to me a good idea to get some results after this operation, I thought to get a string and then convert it into a message on the screen

Reply:
@hardioi the fact that this method have not thrown any exception is itself an indicator of a success. So we can just show a successful message if we didn't get any errors.